
// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option objc_class_prefix = "HLW";

package RpcModule;

// The greeting service definition.
service Rafter {
  // 如果这个为空，则视为心跳包，可以携带信息
  rpc ClientRequest (ClientRequestArgs) returns (ClientRequestReply) {}
  rpc RegisterClient (RegisterClientArgs) returns (RegisterReply) {}
  rpc ClientQuery (ClientQueryArgs) returns (ClientQueryReply) {}
}

//这个是Client请求的
message ClientRequestArgs{
  int32 clientId=1;
  int32 sequenceNum=2;
  string command=3;
}
message ClientRequestReply{
  int32 status=1;
  string response=2;
  int32 leaderHint=3;
}
//这个是RegisterClientArgs请求的
message RegisterClientArgs{

}
message RegisterReply{
  int32 status=1;
  int32 clientId=2;
  int32 leaderHint=3;
}
//这个是ClientQueryArgs请求的
message ClientQueryArgs{
  int32 query=1;
}
message ClientQueryReply{
  int32 status=1;
  string response=2;
  int32 leaderHint=3;
}